lib/src/logger_extension.dart

```diff
@@ -11,13 +21,25 @@ extension LoggerX on Logger {
   /// Wrap the [text] to fit perfectly within the width of the terminal when
   /// [print]ed.
   ///
-  /// To overwrite the width you can use [length].
+  /// The text will wrap around the terminal width, and will not break words. If
+  /// the terminal width cannot be determined, the text will wrap around the
+  /// [fallbackStdoutTerminalColumns].
+  ///
+  /// To completely overwrite the width you can use [length].
   void wrap(
     String? text, {
     required void Function(String?) print,
     int? length,
   }) {
-    final maxLength = length ?? stdout.terminalColumns;
+    late final int maxLength;
+    if (length != null) {
+      maxLength = length;
+    } else if (stdout.hasTerminal) {
+      maxLength = stdout.terminalColumns;
+    } else {
+      maxLength = fallbackStdoutTerminalColumns;
+    }
```

Comment:
```suggestion
    final maxLength = switch (length) {
      == null when stdout.hasTerminal => stdout.terminalColumns,
      == null when !stdout.hasTerminal => fallbackStdoutTerminalColumns,
      _ => length
    };
```

Reply:
I think this is Nit. Tbh, I think the suggestion is less legible. Does it have an impact on performance?

Reply:
Then u add a `!` to it

Reply:
it is guaranteed to be non-null anyway, but because `hasTerminal` is not a constant value it can't figure that out